src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/ContaminationModel.java

```diff
@@ -160,15 +162,49 @@ public ContaminationModel(List<PileupSummary> sites) {
                 .mapToDouble(ps -> ps.getTotalCount() * oppositeAlleleFrequency.applyAsDouble(ps))
                 .sum();
 
-        final double contamination = contaminationOppositeDepth / totalDepthWeightedByOppositeFrequency;
+        final double contaminationEstimate = contaminationOppositeDepth / totalDepthWeightedByOppositeFrequency;
+
+        final double coeff1 = homs.stream().mapToDouble(ps -> oppositeAlleleFrequency.applyAsDouble(ps) * ps.getTotalCount()).sum();
+        final double coeff2 = homs.stream().mapToDouble(ps ->
+                oppositeAlleleFrequency.applyAsDouble(ps)*(1 - oppositeAlleleFrequency.applyAsDouble(ps)) * MathUtils.square(ps.getTotalCount())
+        ).sum();
+
+        final DoubleUnaryOperator errorFunc = c -> homs.isEmpty() ? 1 : Math.sqrt(coeff1*c*(1-c) + coeff2*c*c) / totalDepthWeightedByOppositeFrequency;
+
+        // we're going to binary search to find the largest contamination whose expected standard error brings it within range of
+        // our estimate.  That is, suppose we estimate a contamination of 0.03 and the standard error of 0.05 is 0.02.  Then 0.05 is
```

Comment:
standard error of 0.05

Could you make this clearer?

Reply:
done